REFACTOR ALL THE THINGS

hyperdriveguy opened the pull request:
See #11.

hyperdriveguy committed:
Initial text module

hyperdriveguy committed:
Shoehorn the current text module into the monstrosity

hyperdriveguy committed:
Fix a typo

hyperdriveguy committed:
More progress on moving text

hyperdriveguy committed:
Add bpm to musicxml template

nephitejnf commented:
Looks good so far to my fire ridden brain, would you rather I wait before a merge?

hyperdriveguy commented:
Yeah, this is just the beginning. I'm hoping to fix a bunch of issues as they come up and some of the existing ones as well. I also want to wait so I can test for any possible regressions as I go along as well.

nephitejnf commented:
> I'm hoping to fix a bunch of issues as they come up and some of the **existing ones** as well.

Heh

hyperdriveguy commented on src/text.py:
```diff
@@ -0,0 +1,195 @@
+    Text that gets output to the terminal.
+
+    All the text that stays constant are in one method.
+    Text the more reusable pieces of text are methods.
```
Oops, that's a typo

hyperdriveguy replied:
Fiexed

hyperdriveguy committed:
Started to factor out main() and switched from get getopt to argparse
Usage is now more unix like and short options are finally now supported.

hyperdriveguy commented:
Help output isn't as pretty from argparse, but it's slightly less code and is properly supporting short options as well as allowing us to drop `-i` and `-o` as options

*As I said, it has only begun...*

hyperdriveguy committed:
I know it's legacy, but it's good to have it work properly anyway

hyperdriveguy committed:
Start working on parser

hyperdriveguy committed:
Temporary option mostly for testing

hyperdriveguy committed:
The beginnings of some parser output
Just prints a list to the terminal for now

hyperdriveguy committed:
Command output beginnings
Note and command parsing gets it's own module because it's a thiccc 
function

hyperdriveguy committed:
Finish more of the note parsing module
Mostly done, started some optimizations of rests that may break in very 
specific edge cases.
Also added a flag to toggle optimizations.

hyperdriveguy committed:
Merge branch 'master' of https://github.com/nephitejnf/muse2pokecrystal into refactor
Pull licensing and Readme changes

hyperdriveguy committed:
Module docstring including license information

hyperdriveguy committed:
Very raw and initial copypasta code
Todo: codense the process_to_file_store method by actually using 
conditionals and a loop

hyperdriveguy commented:
I'll consider this ready for review once the new implementation is at least on par with the previous feature set.

# What's left
## Features
- [x] Reimplement `configparser`
- [x] File write
- [x] Reimplement `parity_check` (Wasn't orginally planning on it, but it's a lot more compact than the original implementation.)
- [ ] ~~Ignore notes that are part of a chord and throw a warning (replaces/coincides with the purpose of `parity_check`)~~
  (I can save this for a future PR if you want.)

## Regressions
- [x] Tempo is output twice
## New bugs
- [ ] ~~Rest optimizations may cause desync if relying on proper `notetype` timing on a rest~~ (**Won't Fix**)
## Code readability
- [x] Final touches on `process_to_file_store()`

hyperdriveguy committed:
More refactor progress
Made module docstrings more useful
Prettied up the output_to_file_store method
Started to add custom exceptions (complete move is out of scope for this 
refactor)
Fixed redundant tempo command

hyperdriveguy committed:
Reimplement configuration parsing

hyperdriveguy committed:
File writing function

hyperdriveguy committed:
Reimplement parity check
This is still kind of done the old way.
My hope is have actual chord handling soon in the ParseStaff class.

hyperdriveguy commented:
Do you want me to overwrite the contents of `muse2pokecrystal.py` with that of `refactor.py`?
I'll go ahead and do it, but if you want me to revert it, let me know.

hyperdriveguy committed:
Remove cruft
